lib/segment/src/data_types/vectors.rs

```diff
@@ -68,7 +68,7 @@ impl<'a> TryFrom<VectorRef<'a>> for &'a MultiDenseVector {
 
     fn try_from(value: VectorRef<'a>) -> Result<Self, Self::Error> {
         match value {
-            VectorRef::Dense(_) => Err(OperationError::WrongMulti),
+            VectorRef::Dense(_) => Err(OperationError::WrongMulti), // VectorRef::Dense cannot be converted to &MultiDense
```

Comment:
Why `Vector` can be converted and `VectorRef` cannot be?

Reply:
Here we are implementing `impl<'a> TryFrom<VectorRef<'a>> for &'a MultiDenseVector`

A `VectorRef::Dense` contains a `&[VectorElement]` but `MultiDenseVector` requires a own vector.
This could be solved with a reallocation.

However the final type is `&MultiDenseVector` with an outer lifetime.

```
error[E0515]: cannot return value referencing temporary value
  --> lib/segment/src/data_types/vectors.rs:71:36
   |
71 |             VectorRef::Dense(v) => Ok(&MultiDenseVector::new(v.to_owned(), 1)),
   |                                    ^^^^--------------------------------------^
   |                                    |   |
   |                                    |   temporary value created here
   |                                    returns a value referencing data owned by the current function
```

I can't return a reference to the temporary object.

WDYT?

Reply:
We don't need `&MultiDenseVector`, use `TypedMultiDenseVectorRef<f32>` instead. It allows such conversion. And you can do `impl<'a> From<&'a MultiDenseVector> for TypedMultiDenseVectorRef<'a, f32>`

Reply:
I tried to do it in a separate PR:
https://github.com/qdrant/qdrant/pull/4224
The result is sad. If `VectorRef` is converted into `MultiDenseVectorRef`, the user would use dense vectors as multivectors everywhere like in vector insertion.
On another hand, I cannot approve different conventions from `Vector` and `VectorRef`. It's too unexplicit